api/config.go

```diff
@@ -12,14 +24,150 @@ type ConfigUnset struct {
 
 // Config is the structure of an app's config.
 type Config struct {
-	Owner    string                 `json:"owner,omitempty"`
-	App      string                 `json:"app,omitempty"`
-	Values   map[string]interface{} `json:"values,omitempty"`
-	Memory   map[string]interface{} `json:"memory,omitempty"`
-	CPU      map[string]interface{} `json:"cpu,omitempty"`
-	Tags     map[string]interface{} `json:"tags,omitempty"`
-	Registry map[string]interface{} `json:"registry,omitempty"`
-	Created  string                 `json:"created,omitempty"`
-	Updated  string                 `json:"updated,omitempty"`
-	UUID     string                 `json:"uuid,omitempty"`
+	Owner        string                 `json:"owner,omitempty"`
+	App          string                 `json:"app,omitempty"`
+	Values       map[string]interface{} `json:"values,omitempty"`
+	Memory       map[string]interface{} `json:"memory,omitempty"`
+	CPU          map[string]interface{} `json:"cpu,omitempty"`
+	Healthcheck  map[string]Healthcheck `json:"healthcheck,omitempty"`
+	Tags         map[string]interface{} `json:"tags,omitempty"`
+	Registry     map[string]interface{} `json:"registry,omitempty"`
+	Created      string                 `json:"created,omitempty"`
+	Updated      string                 `json:"updated,omitempty"`
+	UUID         string                 `json:"uuid,omitempty"`
+}
+
+// Healthcheck is the structure for an application healthcheck.
+// Healthchecks only need to provide information about themselves.
+// All the information is pushed to the server and handled by kubernetes.
+type Healthcheck interface {
+	fmt.Stringer
+	Probe() fmt.Stringer
+}
+
+// KVPair is a key/value pair used to parse values from
+// strings into a formal structure.
+type KVPair struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+// ExecProbe executes a command within a Pod.
+type ExecProbe struct {
+	Command []string `json:"command"`
+}
+
+// String displays the ExecProbe in a readable format.
+func (e ExecProbe) String() string {
+	msg := "Command: "
+	for _, cmd := range e.Command {
+		msg += cmd + " "
+	}
+	return strings.TrimSpace(msg)
+}
+
+// HTTPGetProbe performs an HTTP GET request to the Pod
+// with the given path, port and headers.
+type HTTPGetProbe struct {
+	Path        string   `json:"path"`
+	Port        int   `json:"port"`
```

Comment:
port can be a string 
http://kubernetes.io/docs/api-reference/v1/definitions/#_v1_httpgetaction